samples/detect.js

```diff
@@ -770,6 +770,71 @@ function detectFulltextGCS(bucketName, fileName) {
   // [END vision_fulltext_detection_gcs]
 }
 
+function detectPdfText(bucketName, fileName) {
+  // [START vision_async_detect_document_ocr]
+
+  // Imports the Google Cloud client libraries
+  const vision = require('@google-cloud/vision').v1;
+
+  // Creates a client
+  const client = new vision.ImageAnnotatorClient();
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // Bucket where the file resides
+  // const bucketName = 'my-bucket';
+  // Path to PDF file within bucket
+  // const fileName = 'path/to/document.pdf';
+
+  const gcsSourceUri = `gs://${bucketName}/${fileName}`;
+  const gcsDestinationUri = `gs://${bucketName}/${fileName}.json`;
+
+  const inputConfig = {
+    // Supported mime_types are: 'application/pdf' and 'image/tiff'
+    mimeType: 'application/pdf',
+    gcsSource: {
+      uri: gcsSourceUri,
+    },
+  };
+  const outputConfig = {
+    gcsDestination: {
+      uri: gcsDestinationUri,
+    },
+  };
+  const features = [{type: 'DOCUMENT_TEXT_DETECTION'}];
+  const request = {
+    requests: [
+      {
+        inputConfig: inputConfig,
+        features: features,
+        outputConfig: outputConfig,
+      },
+    ],
+  };
+
+  client
+    .asyncBatchAnnotateFiles(request)
+    .then(results => {
+      const operation = results[0];
+      // Get a Promise representation of the final result of the job
+      operation
```

Comment:
You don't need to nest your promises :) . Just return `operation.promise()` and use promise chaining to keep things more flat:

```js
.then(results => {
  const operation = results[0];
  return operation.promise();
}).then(filesResponse => {
  // do work
}).catch(console.error);
```